service/SpaceCenter/src/Services/Orbit.cs

```diff
@@ -362,5 +362,138 @@ public double OrbitalSpeedAt (double time)
+        [KRPCMethod]
+        public double RadiusAt(double time)
+        {
+            return InternalOrbit.getRelativePositionAtUT(time).magnitude;
+        }
+
+        /// <summary>
+        /// The orbital radius at the given time, in meters.
+        /// </summary>
+        /// <param name="time">Time from now, in seconds.</param>
+        /// <param name="rf">Reference Frame.</param>
+        [KRPCMethod]
+        public Tuple3 PositionAt(double time, ReferenceFrame rf)
+        {
+            return rf.PositionFromWorldSpace(InternalOrbit.getPositionAtUT(time)).ToTuple();
+
+        }
+
+
+        /// <summary>
+        /// Estimates time of closest approach in the next orbit.
+        /// </summary>
+        /// <param name="target">Other vessel.</param>
+        [KRPCMethod]
+        public double ClosestApproach(Vessel target)
+        {           
+            Orbit b = target.Orbit;
+            double approachTime = Planetarium.GetUniversalTime();
+            double approachDistance = double.MaxValue;
+            double mintime = Planetarium.GetUniversalTime();
+            double interval = InternalOrbit.period;
+            if (InternalOrbit.eccentricity > 1.0) { interval = 100 / InternalOrbit.meanMotion; }
+            double maxtime = mintime + interval;
+
+            //Conduct coarse search
+            double timestep = (maxtime - mintime) / 20;
+            double placeholder = mintime;
+            while (placeholder < maxtime)  
+            {
+                Vector3d PosA = InternalOrbit.getPositionAtUT(placeholder);
+                Vector3d PosB = b.InternalOrbit.getPositionAtUT(placeholder);
+                double thisDistance = Vector3d.Distance(PosA, PosB);
+                if (thisDistance < approachDistance)
+                {
+                    approachDistance = thisDistance;
+                    approachTime = placeholder;
+                }
+                placeholder += timestep;
+            }
+
+            //Conduct fine search
+            double fine_mintime = approachTime - timestep;
+            double fine_maxtime = approachTime + timestep;
+            if (fine_maxtime > maxtime) fine_maxtime = maxtime;
+            if (fine_mintime < mintime) fine_mintime = mintime;
+            timestep = (fine_maxtime - fine_mintime) / 50;
+            placeholder = fine_mintime;
+
+            while (placeholder < fine_maxtime)
+            {
+                Vector3d PosA = InternalOrbit.getPositionAtUT(placeholder);
+                Vector3d PosB = b.InternalOrbit.getPositionAtUT(placeholder);
+                double thisDistance = Vector3d.Distance(PosA, PosB);
+                if (thisDistance < approachDistance)
+                {
+                    approachDistance = thisDistance;
+                    approachTime = placeholder;
+                }
+                placeholder += timestep;
+            }
+            return approachTime;
+        }
+
+
+        /// <summary>
```

Comment:
That all makes perfect sense!  I'll jump right on it!  Yes, the code for the time and distance is identical, because you have to find one to find the other.  Initially I returned them as a tuple (time, distance) but since it wasn't a named tuple it wasn't very self explanatory.  I'll put it into a helper function and let the two KRPCMethods grab what they need to from it!  

It also occurs to me that it might be useful to be able to get the closest approach for the next several orbits?  Does that seem worth adding?   Would you add it as a separate but similar function that took a 'number of orbits' parameter and returned a list of values...  or would you change these existing functions to take a 'number of orbits' that defaults to 1.0, and always returns a list, even if it's just one value?

I am also trying to figure out the last bit of rendezvous math...  some sort of lambert solver/estimator for timing the hohmann transfer.  I tried the hohmann function in the krpctoolkit directory, but it wasn't getting particularly close to the correct timing?  I've got an idea for the algorithm, but wanted your guidance - is something like that too 'high level' to build into KRPC?  Is it better to just expose whatever data is required for it and then let users do the calculations in their client?  Or would a 'CreateInterceptNode(target)' function make sense?